app/ui-react/packages/api/src/WithUserHelpers.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { ApiContext, IApiContext } from './ApiContext';
+import { callFetch } from './callFetch';
+
+export interface IWithUserHelpersChildrenProps {
+  logout(): Promise<any>;
+}
+export interface IWithUserHelpersProps {
+  children(props: IWithUserHelpersChildrenProps): any;
+}
+
+export class WithUserHelpersWrapped extends React.Component<
+  IWithUserHelpersProps & IApiContext
+> {
+  constructor(props: IWithUserHelpersProps & IApiContext) {
+    super(props);
+    this.logout = this.logout.bind(this);
+  }
+
+  public async logout() {
+    const response = await callFetch({
+      headers: {
+        ...this.props.headers,
+        responseType: 'arraybuffer',
```

Comment:
```suggestion
        'SYNDESIS-XSRF-TOKEN': 'awesome',
        responseType: 'arraybuffer',
```

Reply:
That's not needed, the right headers come from [here](https://github.com/syndesisio/syndesis-react/pull/332/files/19ef60977804468b9d366ed71d52c208000946b6#diff-abcbbbbc168c9e47faf84ef500912928R49)

Reply:
Yep, I do see `SYNDESIS-XSRF-TOKEN` being provided as part of the request headers 👍